src/app/shared/entry.ts

```diff
@@ -461,6 +461,16 @@ export abstract class Entry implements OnDestroy {
    * @param topicId The ID of the topic on discourse
    */
   discourseHelper(topicId: number): void {
+    // If the #discourse-comments element does not exist, or if the previous topicId
+    // (from the previous call) is the same as the requested topicId, no dothing.
+    const comments = document.getElementById('discourse-comments');
+    if (!comments || (<any>comments).topicId === topicId) {
```

Comment:
The more TypeScript-y way would be to use `setAttribute` and `getAttribute`. It's our goal one day to get rid of `any` and to avoid introducing new usages of it in the meantime. Although we use it so much, I don't know if we ever will.

If you do change it, note that `setAttribute` converts the value to a string :(

Reply:
I understand the desire to get of `any`, but in this case, I literally want to read/write a property from/to the raw Javascript object, rather than set an attribute.